pkg/stanza/operator/transformer/recombine/recombine.go

```diff
@@ -315,19 +318,20 @@ func (r *Transformer) addToBatch(_ context.Context, e *entry.Entry, source strin
 // flushUncombined flushes all the logs in the batch individually to the
 // next output in the pipeline. This is only used when there is an error
 // or at shutdown to avoid dropping the logs.
-func (r *Transformer) flushUncombined(ctx context.Context) {
+func (r *Transformer) flushUncombined(ctx context.Context) error {
 	for source := range r.batchMap {
-		for _, entry := range r.batchMap[source].entries {
-			r.Write(ctx, entry)
+		err := r.flushSource(source, true, ctx)
+		if err != nil {
+			return err
 		}
-		r.removeBatch(source)
 	}
 	r.ticker.Reset(r.forceFlushTimeout)
+	return nil
 }
 
 // flushSource combines the entries currently in the batch into a single entry,
 // then forwards them to the next operator in the pipeline
-func (r *Transformer) flushSource(source string, deleteSource bool) error {
+func (r *Transformer) flushSource(source string, deleteSource bool, ctx context.Context) error {
```

Comment:
This fails lint:
`Error: stanza/operator/transformer/recombine/recombine.go:334:69: context-as-argument: context.Context should be the first parameter of a function (revive)`